js/components/icon.js

```diff
@@ -0,0 +1,28 @@
+var React = require('react');
+
+
+var Icon = React.createClass({
+  shouldComponentUpdate: function(nextProps, nextState) {
```

Comment:
Player was getting unresponsive re-rendering a ton of icons for no reason. I'm guessing react doesn't intelligently compare objects inside of props, hence the naive stringify comparison here. This removed all the extra renders according to the React perf tool.